Building instructions

Please could the build process be given/checked?  Please reuse anything below if useful...

I have had issues building (after a 35 year lapse in writing C/C++).  So far I have done the following:

1. Installed Docker Desktop (I'm using windows) - following this guide https://community.element14.com/products/raspberry-pi/b/blog/posts/working-with-pi-pico-c-c-projects-easily-a-docker-approach
2. Checked out the project using `git clone https://github.com/Loc15/PicoGamepadConverter.git` into /home/dev/pico
3. Entered these commands - last command gave an **Error for Pico-PIO-USB**

```
mkdir build  
cd build  
cmake ../src
```

4. So then did

```
cd ..
git clone https://github.com/sekigon-gonnoc/Pico-PIO-USB.git Pico-PIO-USB
git clone https://github.com/hathach/tinyusb tinyusb
```

5. Tried again:

```
cd build  
cmake ../src
```

- This gave an error:

```
CMake Error at CMakeLists.txt:25 (add_executable):
  Cannot find source file:

    /home/dev/pico/PicoGamepadConverter/src/device_files/rndis-lib/mongoose/mongoose.c

  Tried extensions .c .C .c++ .cc .cpp .cxx .cu .mpp .m .M .mm .ixx .cppm .h
  .hh .h++ .hm .hpp .hxx .in .txx .f .F .for .f77 .f90 .f95 .f03 .hip .ispc
```

6. I then tried dropping the mongoose.c and .h into the src - which did allow cmake to finish,  but then using make (to create the UF2) fails with:

```
[100%] Linking CXX executable PicoGamepadConverter.elf
/usr/lib/gcc/arm-none-eabi/12.2.0/../../../../arm-none-eabi/bin/ld: CMakeFiles/PicoGamepadConverter.dir/device_files/rndis-lib/mongoose/mongoose.c.obj: in function `p_mkdir':
mongoose.c:(.text.p_mkdir+0x6): undefined reference to `mkdir'
Memory region         Used Size  Region Size  %age Used
           FLASH:      498172 B         2 MB     23.75%
             RAM:       50032 B       256 KB     19.09%
       SCRATCH_X:          2 KB         4 KB     50.00%
       SCRATCH_Y:          0 GB         4 KB      0.00%
collect2: error: ld returned 1 exit status
make[2]: *** [CMakeFiles/PicoGamepadConverter.dir/build.make:3271: PicoGamepadConverter.elf] Error 1
make[1]: *** [CMakeFiles/Makefile2:1678: CMakeFiles/PicoGamepadConverter.dir/all] Error 2
make: *** [Makefile:91: all] Error 2
```

--- a/README.md
+++ b/README.md
@@ -144,7 +144,7 @@ Controllers that was tested on different host modes.
 _The instructions below have been successful on Windows (11) but should mostly apply to Linux_
 
 1. (windows) Install Docker Desktop
-2. Setup pico sdk environment by following https://community.element14.com/products/raspberry-pi/b/blog/posts/working-with-pi-pico-c-c-projects-easily-a-docker-approach
+2. Setup pico sdk environment by following  [this guide](https://community.element14.com/products/raspberry-pi/b/blog/posts/working-with-pi-pico-c-c-projects-easily-a-docker-approach)
 3. Open Docker Desktop
 4. Find the `pc1` container, click on `⋮` and select `Open in Terminal` then type in these commands:
 
@@ -178,8 +178,6 @@ docker cp pc1:/home/dev/pico/PicoGamepadConverter/build/PicoGamepadConverter.uf2
 cp PicoGamepadConverter.uf2 d:
 ```
 9. Start using...
-
-_You can also use VSCode to edit the container code - install the extension  `Remote Development`.  Then select the left hand icon `Remote Explorer` (this might need installing) and then select the local container.  Now you can use VS Code for editing and the terminal is in the container as well_
 
 ---
 

--- a/README.md
+++ b/README.md
@@ -139,6 +139,50 @@ Controllers that was tested on different host modes.
 
 ---
 
+### Building
+
+_The instructions below have been successful on Windows (11) but should mostly apply to Linux_
+
+1. (windows) Install Docker Desktop
+2. Setup pico sdk environment by following https://community.element14.com/products/raspberry-pi/b/blog/posts/working-with-pi-pico-c-c-projects-easily-a-docker-approach
+3. Open Docker Desktop
+4. Find the `pc1` container, click on `⋮` and select `Open in Terminal` then type in these commands:
+
+```
+cd /home/dev
+mkdir pico
+cd pico
+git clone --recursive https://github.com/Loc15/PicoGamepadConverter.git
+```
+5. Wait a bit then:
+
+```
+cd PicoGamepadConverter
+mkdir build
+cd build
+cmake ../src
+```
+
+6. Wait a bit more then:
+
+```
+make
+```
+
+7. Wait some more, then reset your Pico (or other board) with BOOTSEL held down - the instructions below assume the Pico will be drive D:
+8. Open a **windows command line** (i.e. windows powershell - NOT the docker terminal) and type in:
+
+```
+cd Downloads
+docker cp pc1:/home/dev/pico/PicoGamepadConverter/build/PicoGamepadConverter.uf2 .
+cp PicoGamepadConverter.uf2 d:
+```
+9. Start using...
+
+_You can also use VSCode to edit the container code - install the extension  `Remote Development`.  Then select the left hand icon `Remote Explorer` (this might need installing) and then select the local container.  Now you can use VS Code for editing and the terminal is in the container as well_
+
+---
+
 ### Troubleshooting
 
 - 8bitdo controllers sometimes have problems to connect on USB or Bluetooth mode. On USB if it doesn't connect reboot the microcontroller without disconnect. If it doesn't work either, [reset pico's flash](https://www.raspberrypi.com/documentation/microcontrollers/raspberry-pi-pico.html#resetting-flash-memory). 

